woocommerce-services.php

```diff
@@ -530,13 +530,29 @@ public function init_core_wizard_shipping_config() {
 
 		public function init_core_wizard_payments_config() {
 			$stripe_settings = get_option( 'woocommerce_stripe_settings', false );
-			$stripe_enabled  = is_array( $stripe_settings )
+
+			$user_elected_to_create_stripe_account  = is_array( $stripe_settings )
 				&& ( isset( $stripe_settings['create_account'] ) && 'yes' === $stripe_settings['create_account'] )
 				&& ( isset( $stripe_settings['enabled'] ) && 'yes' === $stripe_settings['enabled'] );
 
-			if ( $stripe_enabled && is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' ) ) {
+			// In certain scenarios, the user can enter an email address but not connect Jetpack in the wizard,
+			// but instead add the Stripe keys manually and connect Jetpack after. If the existing keys are detected,
+			// forget the wizard settings and never retry
+			$stripe_already_connected = is_array( $stripe_settings )
+				&& (
+					! empty( $stripe_settings['test_publishable_key'] )
+					|| ! empty( $stripe_settings['test_secret_key'] )
+					|| ! empty( $stripe_settings['publishable_key'] )
+					|| ! empty( $stripe_settings['secret_key'] )
+				);
+			if ( $user_elected_to_create_stripe_account && $stripe_already_connected ) {
```

Comment:
We're currently tracking the event when the account is created:

`$this->tracks->record_user_event( 'core_wizard_stripe_setup' );`

in https://github.com/Automattic/woocommerce-services/blob/d8cc3858a49e5a592786c61a56b6fd709ca5dd7a/woocommerce-services.php#L556

Should we track when we tried to set up the account but are not able to, for example using

```
$this->tracks->record_user_event( 'core_wizard_stripe_setup_failed' );
```

Reply:
I just double-checked about the best way to track this event. We should create a new event and have `success` and `reason` props, something like this would work for this "not tried" case:

$this->tracks->record_user_event( 'core_wizard_stripe_setup_failed', array(
  'success' => false,
  'reason' => 'already set up by user'
) );

And for the failing otherwise:

$this->tracks->record_user_event( 'core_wizard_stripe_setup_failed', array(
  'success' => false,
  'reason' => 'email already exists'
) );

$this->tracks->record_user_event( 'core_wizard_stripe_setup_failed', array(
  'success' => false,
  'reason' => $wp_error
) );

And success:

$this->tracks->record_user_event( 'core_wizard_stripe_setup_failed', array(
  'success' => true,
) );

I didn't test out the array, but it looks like it should work.

Reply:
Btw, I think we can just fix the success tracking in this case with a new event name, be able to release this today, and then have an issue for the optimal tracking down the road to be released later. What do you think @marcinbot ?

Reply:
Yep, happy to move this to a different PR. Can you create an issue so we don't forget?

Fwiw, we're tracking both failures and successes in one event, which is why we should at least correct that behaviour.

As for tracking failures when the user has already added their keys - that shows whether or not the features is working as intended. The goal would be for those events to be near 0 at all times. If they're not, then it tells us that something is preventing our users from using the feature to its full potential.